### PR TITLE
test: add unit tests for UserServiceImpl and increase coverage for #199

### LINKED ISSUE #199: Add unit tests to UserController and UserServiceImpl
Review the UserController at `com/n1netails/n1netails/api/controller/UserController.java`
and the UserServiceImpl at `com/n1netails/n1netails/api/service/impl/UserServiceImpl.java`

Implement unit tests for each of the endpoints in the UserController and UserServiceImpl.

### PR DESCRIPTION
**Pull request by Whxismou1**
# 🚀 Pull Request

## 📌 Summary
This PR adds comprehensive unit tests for the UserServiceImpl class.

## 🧪 What Does This Fix or Improve?
- [ ] Bug fix
- [X] New feature: Unit Testing
- [ ] Performance improvement
- [ ] Documentation update
- [ ] Other

## 🔍 Changes
Implemented new unit tests for the following service methods:

- loadUserByUsername: Verified user retrieval, login date updates, and account locking logic.

- register: Validated the creation of new users, including the "First User" (SUPER_ADMIN) logic and organization initialization.

- updateUserRole: Tested the authority mapping for various roles (HR, Manager, Admin).

- updatePassword & completeTutorial: Verified successful updates and exception handling.

- forgotPasswordRequest: Validated reset request creation and email triggering.

- Improved Coverage: Specifically addressed edge cases like empty passwords during registration and various role-based branches in switches.

## ✅ Checklist
Please ensure the following:

- [X] Code compiles without errors
- [X] Tests pass successfully
- [X] No breaking changes introduced
- [ ] Added or updated documentation if needed
- [X] Branch is up to date with `main`

## 🧪 Testing Steps
To run and verify the newly added tests, use: mvn test -Dtest=UserServiceImplTest

## 📸 Screenshots (if applicable)
Attach images or logs here.

## 🔗 Related Issues
Closes #199

### CODE REVIEW

**Comment by shahidfoy** [comment]
@Whxismou1 thanks for the pr I will look into reviewing this when I get some time. If I merge this pr in would you be interested in receiving 1 million n1netails crypto coin as a gift?

File: n1netails-api/src/test/java/com/n1netails/n1netails/api/service/impl/UserServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+    @Test
+    public void testEditUser_EmailAlreadyTakenByAnotherUser_ShouldThrowException() {
+        UsersEntity existingUserInDb = new UsersEntity();
+        existingUserInDb.setId(1L);
+        existingUserInDb.setEmail("original@test.com");
+
+        UsersEntity anotherUserInDb = new UsersEntity();
+        anotherUserInDb.setId(2L);
+        anotherUserInDb.setEmail("taken@test.com");
+
```
Comment (shahidfoy) [inline n1netails-api/src/test/java/com/n1netails/n1netails/api/service/impl/UserServiceImplTest.java:376]: looks like this test is incomplete. Do we still need this test lmk

**Review by shahidfoy — CHANGES_REQUESTED** [review]
looks good but one of the tests is incomplete. please take a look at it. Also make sure to update the pom versions.

(root) pom.xml - 0.3.29 -> 0.3.30
n1netails-api/pom.xml - 0.3.23 -> 0.3.24

File: n1netails-api/src/test/java/com/n1netails/n1netails/api/service/impl/UserServiceImplTest.java (hunk 1 above)
Comment (shahidfoy) [reply n1netails-api/src/test/java/com/n1netails/n1netails/api/service/impl/UserServiceImplTest.java:376]: lets make this test happen it is a good check

Comment (Whxismou1) [reply n1netails-api/src/test/java/com/n1netails/n1netails/api/service/impl/UserServiceImplTest.java:376]: Done! I've implemented the test case to verify email collisions.

I decided to trigger this check through the register method instead of editUser, because register already implements the validateEmail logic. I wanted to avoid modifying the existing production logic in UserServiceImpl just for the sake of the test XD.

Let me know if you agree with this approach! 
Thanks.

**Comment by shahidfoy** [comment]
@Whxismou1 this looks great I will look into reviewing it later  today and hopefully get it merged in

**Comment by shahidfoy** [comment]
@Whxismou1 thanks for your pr I will merge it in let me know if you would like to receive 1 million n1netails token. if yes send me your public SOL address and I'll send it your way

**Review by shahidfoy — APPROVED** [review]

**Comment by Whxismou1** [comment]
> @Whxismou1 thanks for your pr I will merge it in let me know if you would like to receive 1 million n1netails token. if yes send me your public SOL address and I'll send it your way

Hi! I’m happy to hear the PR was merged.
Thank you for offering the token reward — much appreciated.
Here is my Solana public address:
6NwqhuepkCihG3Rh1PHarFzQs8aEnWFNpqryG5PkMQxL

**Comment by shahidfoy** [comment]
thanks @Whxismou1 I sent 1 million n1netails token your way. Please free free to look into other issues if you are interested
you can also join our discord here
https://discord.gg/ma9CCw7F2x
